Microsoft.HealthVault.Fhir/Transformers/PersonItemToFhir.cs

```diff
@@ -0,0 +1,170 @@
+                   }
+                };
+            }
+
+            if (person.ContactInformation != null)
+            {
+                practitioner.Address.AddRange(
+                    person.ContactInformation.Address.Select(address => ToFhirInternal(address)));
+                practitioner.Telecom.AddRange(
+                    person.ContactInformation.Phone.Select(phone => ToFhirInternal(phone)));
+                practitioner.Telecom.AddRange(
+                    person.ContactInformation.Email.Select(email => ToFhirInternal(email)));
+            }
+
+            return practitioner;
+        }
+
+        private static HumanName ToFhirInternal(Name hvName)
+        {
+            var fhirName = new HumanName
+            {
+                Text = hvName.Full,
+                Family = hvName.Last
+            };
+
+            if (hvName.Title != null)
+            {
+                fhirName.AddExtension(HealthVaultExtensions.PatientTitle, hvName.Title.ToFhir());
+            }
+
+            AddGivenName(fhirName, hvName.First);
+            AddGivenName(fhirName, hvName.Middle);
+
+            if (hvName.Suffix != null)
+            {
+                fhirName.AddExtension(HealthVaultExtensions.PatientSuffix, hvName.Suffix.ToFhir());
+            }
+
+            return fhirName;
+        }
+
+        private static void AddGivenName(HumanName fhirName, string first)
+        {
+            if (!string.IsNullOrEmpty(first))
+            {
+                fhirName.GivenElement.Add(new FhirString(first));
+            }
+        }
+
+        private static Hl7.Fhir.Model.Address ToFhirInternal(ItemTypes.Address hvAddress)
+        {
+            var address = new Hl7.Fhir.Model.Address
+            {
+                Text = hvAddress.Description,
+                Line = hvAddress.Street,
+                City = hvAddress.City,
+                PostalCode = hvAddress.PostalCode,
+                District = hvAddress.County,
+                State = hvAddress.State,
+                Country = hvAddress.Country,
+            };
+
+            if (hvAddress.IsPrimary.HasValue && hvAddress.IsPrimary.Value)
+            {
+                address.AddExtension(HealthVaultExtensions.IsPrimary, new FhirBoolean(true));
+            }
+
+            return address;
+        }
+
+        private static ContactPoint ToFhirInternal(Phone phone)
+        {
+            var contactPoint = new ContactPoint
+            {
+                System = ContactPoint.ContactPointSystem.Phone,
+                Value = phone.Number,
+                Rank = phone.IsPrimary.HasValue && phone.IsPrimary.Value ? 1 : (int?)null,
+            };
+
+            if (!string.IsNullOrEmpty(phone.Description))
+            {
+                contactPoint.AddExtension(HealthVaultExtensions.Description, new FhirString(phone.Description));
+            }
+
+            return contactPoint;
+        }
+
+        private static ContactPoint ToFhirInternal(Email email)
```

Comment:
Can you abstract this out to where it can also be used in "Transformers/ContactToFhir.cs"? There's very similar code there.

Reply:
Fixed; Added a new ToFhir extension for ItemTypes.Email and the same is used in both PersonItem and Contact transformer classes